src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -758,28 +772,32 @@ public IAsyncOperation<ComputeSystemOperationResult> DoPinActionAsync(string loc
     public IAsyncOperation<ComputeSystemOperationResult> PinToStartMenuAsync()
     {
         // Since there is no 'Pinning' state we'll say that the state is saving
-        StateChanged?.Invoke(this, ComputeSystemState.Saving);
+        _hasPinOperationStarted = true;

```

Comment:
Can this line be moved to DoPinActionAsync and undo the next line back to "StateChanged?.Invoke(this, ComputeSystemState.Saving);"?
It seems excessive that we need to set it in every "Pin" method. Having set and reset in one method makes it easier to read and maintain.

<nit>: Naming: Having “Started” in the name often implies that we also interested to know “has operation ended”. But we only want to know “isPinOperationInProgress” here.

Reply:
fair. If we think its too excessive we could move the `StateChanged?.Invoke(this, GetState());` and the `_hasPinOperationStarted` boolean or whatever we call it to be directly used in the `DoPinActionAsync` method. Then we'd only need to have the one `StateChanged` call and then change the `_hasPinOperationStarted` when it starts and when it ends

Reply:
Moved the code; Renamed. Also, Branden has also coded it as such that one pinning operation will need to be done before another starts, so we don't need a counter.

Reply:
> one pinning operation will need to be done before another starts, so we don't need a counter

In general, an extension shouldn't make any assumptions about host (UX) behavior and just do the right thing. Host code can change independently from extensions, have bugs, a completely new host can be created, etc.